### ISSUE TITLE
[FR]: Converting title spaces to dashes for filename

### ISSUE THREAD

**Issue by ennui2342** [ObsidianPublisher/obsidian-github-publisher#60]
### Issue validation

- [X] My issue is not a question relative to [Material Mkdocs](https://squidfunk.github.io/mkdocs-material/) nor [Mkdocs](https://www.mkdocs.org/) (*[You can open a discussion to get more help!](https://github.com/squidfunk/mkdocs-material/discussions)*)
- [X] I checked the issue to prevent duplicate

### Workflow

Obsidian plugin 💠

### What solution do you want to see ?

Great plugin, thanks for making it available.

I'm publishing from Obsidian to a Github pages Jekyll blog. Filenames need the following:

> The document's base filename, with every sequence of spaces and non-alphanumeric characters replaced by a hyphen.

At the moment I'm achieving this with this kind of frontmatter:

```
---
publish: true
filename: 2020-10-01-the-four-stages-of-the-product-lifecycle
---
```

But it would be better if I was able to specify that when the markdown title is used the escaping is done to make it valid for Jekyll. Thanks.

-Mark

### OS

MacOS

**Comment by Lisandra-dev** [comment]
Good idea, i will wrote that :D

**Comment by ennui2342** [comment]
Just trying to test this out but get 'plugin failed to install' when I try to update to 4.9. Any ideas as I have no access to any logs or other information?

**Comment by Lisandra-dev** [comment]
The build fail :'D No worry, I fix that

**Comment by ennui2342** [comment]
4.9.1 all works. Thanks for fixing this.

For anyone looking at this thread in the future, this is the regex I've used for jekyll compatibility:

<img width="865" alt="image" src="https://user-images.githubusercontent.com/912310/209970587-7cb0f389-765d-4d67-a0ad-e8d88fed3e41.png">

Lisandra - One improvement that could be made is to allow the regex to work even if you aren't using the front matter to set the title. If this were possible I could remove the filename front matter - I could just use the name of the note and be sure it would get changed by the regex.

**Comment by Lisandra-dev** [comment]
Oh yes, I agree! I totally forgot that.
I will add this today /)

**Comment by Lisandra-dev** [comment]
Fixed in 4.9.2
